trieste/models/gpflow/interface.py

```diff
@@ -91,3 +92,11 @@ def optimize(self, dataset: Dataset) -> None:
         :param dataset: The data with which to optimize the `model`.
         """
         self.optimizer.optimize(self.model, dataset)
+
+    def reparam_sampler(self, num_samples: int) -> ReparametrizationSampler:
```

Comment:
what about having a generic `sampler` method and you have to pass a `ModelSampler` object? 

depending on the need a user could then pass either an RFF based sampler or reparametrization based method or something else, if you don't use acq functions that need these then you don't pass anything

defining a method for each type  of a sampler does not seem like a great solution, is there a problem with decoupling model and sampler implementation like this?

Reply:
also, if you have this sampler, why not `trajectory_sampler` as well?